docs/root/api-docs/xds_protocol.rst

```diff
@@ -427,13 +427,17 @@ names becomes empty, that means that the client is no longer interested in any r
 specified type.
 
 For :ref:`Listener <envoy_v3_api_msg_config.listener.v3.Listener>` and :ref:`Cluster <envoy_v3_api_msg_config.cluster.v3.Cluster>` resource
-types, there is also a "wildcard" mode, which is triggered when the initial request on the stream
-for that resource type contains no resource names. In this case, the server should use
-site-specific business logic to determine the full set of resources that the client is interested
-in, typically based on the client's :ref:`node <envoy_v3_api_msg_config.core.v3.Node>` identification. Note
-that once a stream has entered wildcard mode for a given resource type, there is no way to change
-the stream out of wildcard mode; resource names specified in any subsequent request on the stream
-will be ignored.
+types, there is also a "wildcard" mode, which comes in two equivalent flavors. First flavor,
```

Comment:
As per my last comment in #15857, I don't think we should be describing this as two separate modes.  Instead, I think we should change this doc to explain that subscribing to `*` means a wildcard subscription, and that if the first request on a stream for a given resource type specifies an empty set of subscriptions, that's a syntactic alias for subscribe to `*` and nothing else.

Reply:
Please have a look at the rewording. Hopefully it's closer to what you wanted.

Reply:
Not quite.  I've sent you #16861 with what I'm proposing.